docs/tutorials/pkg-map.md

```diff
@@ -29,6 +29,23 @@ nixpkgs.overlays = [
 ];
 ```
 
+The user can map package(s) in Nixpkgs to a `pkgconfig-depends` name by
+overlaying the `haskell-nix.extraPkgconfigMappings` attribute:
+
+```nix
+nixpkgs.overlays = [
+  (self: super: {
+    haskell-nix = super.haskell-nix // {
+      extraPkgconfigMappings = super.haskell-nix.extraPkgconfigMappings // {
```

Comment:
You have to take care to not clobber other overlays that add mappings.

I have some helpers in my Nix code since this gets tedious:

```nix
{
    # See https://stackoverflow.com/a/54505212
    recursiveMerge = x: y: recursiveMergeAll [x y];
    recursiveMergeAll = attrList:
      with super.lib;
      let f = attrPath:
            zipAttrsWith (n: values:
              if tail values == []
              then head values
              else if all isList values
              then unique (concatLists values)
              else if all isAttrs values
              then f (attrPath ++ [n]) values
              else last values
            );
      in f [] attrList;

    mapExtraPkgconfig = super-haskell-nix: name: mappings: recursiveMerge super-haskell-nix {
      extraPkgconfigMappings = { "${name}" = mappings; };
    };
}
```

So my `extraPkgconfigMappings` overlays end up looking like this:

```nix
self: super:
{
  haskell-nix = super.macaroniLib.mapExtraPkgconfig super.haskell-nix "freeglut" ["freeglut"];
}
```